### PR TITLE
Add .isJS API

### PR DESCRIPTION
**Pull request by bmeck**
Not in MIME sniffing standard, but in https://html.spec.whatwg.org/multipage/scripting.html#scriptingLanguages , unsure if it fits here but I assume so.

### CODE REVIEW

File: lib/mime-type.js [hunk 1]
```diff
@@ -68,6 +68,37 @@ module.exports = class MIMEType {
     return serialize(this);
   }
 
+  isJS() {
```
Comment (TimothyGu) [inline lib/mime-type.js]: This should check that there are no parameters present, as they are prohibited by the HTML Standard.

Comment (bmeck) [reply lib/mime-type.js]: thats not true, see https://github.com/whatwg/html/pull/3205 , the only restriction is on the `type` parameter of a `<script>` tag, which does not map to MIME (hence use of `type=module`).

Comment (TimothyGu) [reply lib/mime-type.js]: The linked PR only applies to Content-Type of the response, not e.g. the HTML `type` attribute. As it is, none of the [JavaScript MIME types](https://html.spec.whatwg.org/multipage/scripting.html#javascript-mime-type) contain parameters.

Comment (bmeck) [reply lib/mime-type.js]: > For external JavaScript resources, MIME type parameters in `Content-Type` headers are generally ignored. (In some cases the `charset` parameter has an effect.)

Comment (domenic) [reply lib/mime-type.js]: For the purposes of this module, which is used by jsdom, we should stick to the actually-defined JS MIME type list, not the vague Content-Type behavior that is outside the scope of HTML.

File: lib/mime-type.js [hunk 2]
```diff
@@ -68,6 +68,37 @@ module.exports = class MIMEType {
...
+    switch (this._type) {
+      default: return false;
```
Comment (domenic) [inline lib/mime-type.js]: I'm surprised our ESLint doesn't catch this, but stylistically: default last; use curly braces around all blocks; indent all blocks.

Comment (bmeck) [reply lib/mime-type.js]: I think I've fixed this, unsure one what you consider a block for multiple case clauses.

File: lib/mime-type.js (hunk 1 above)
Comment (bmeck) [reply lib/mime-type.js]: @domenic doesn't https://html.spec.whatwg.org/multipage/webappapis.html#fetch-a-single-module-script also check for the MIME, not just string matches? My usage and reason for this PR is a service worker trying to detect if a Module is coming over the wire not parsing script tag attributes.

Comment (domenic) [reply lib/mime-type.js]: Yeah, I guess the HTML spec uses both checks. I suppose we should have an `{ excludeParameters = false } = {}` parameter list for this.

Comment (bmeck) [reply lib/mime-type.js]: I'm open to renaming, but was confused on the exact nature of it being named `excludeParameters` excluding it from the check or ensuring they were no parameters. I've called it `allowParameters` in a new change.

**Pull request merged**